### DISCUSSION TITLE
How to calculate similarity between the two graphs?

### DISCUSSION THREAD

**Discussion by smith-co** [danielegrattarola/spektral#382] [Q&A]
I have to learn similarity between graphs using deep learning. I have many samples (~500k) of graphs. Graphs have ~5000 nodes and ~4000 edges in the average.

How can I compute similarity score between two graphs? I am thinking:

- convert graphs into vectors using Graph2Vec embedding
- then compare them using various similarity calculating techniques like cosine similarity.

I would really appreciate if I can get some feedback whether this is the correct way to approach this problem or not.

**Comment by danielegrattarola** [comment]
Moved this to the Discussions section as it is an open-ended question. 

Your approach looks reasonable to me, did you have a particular doubt?

**Reply by smith-co** [reply]
Thanks for your response @danielegrattarola. I am not expert in the GNN domain. 

So would be curious to know how would Graph2Vec embedding would work instead of generating the embedding using GCN or GAT as an example.

**Comment by dylanfox5** [comment]
Hello - is there any example of this use case? I find myself in a similar situation here. Any further guidance would be appreciated. Thanks!
